### DISCUSSION TITLE
رصد للواقع ومقترحات

### DISCUSSION THREAD

**Discussion by assawalhy** [scicave/javascript-in-arabic#4] [أفكار]
<div dir=rtl>

لقد تصدرت اللغة الإنجليزية في كل مجالات الحياة بعدما كانت لغة كتاب الله هي اللؤلؤة المتألقة والمبتغى لكل طالب علم، فقد انعكست القاعدة وتبدلت الأوضاع. وأظن أن ما نشر تلك اللغة هي الحروب والغزو الذي قامت به بريطانيا أو المملكة المتحدة، وما تبعها من الهيمنة الأمريكية.

فقد وجب علينا نحن المسلمين عموما (والعرب خصوصا) أن نعيد منارة العلم على بحر اللغة العربية وأهمها العلوم التقنية.

## كيف نعيد تشغيل منارة العلم

- مما يسرع هذا هو وجود دولة تدعم البحث العلمي الذي يصدر إجبارا باللغة العربية وينشر في مواقع ومجلات عربية ووضع بعض النسخ المترجمة ووضع محتوى بما يكفي فقط لجذب انتباه بقية العلماء للأبحاث فمثلا يوضع الملخص والنتائج للبحث. فلو استطعنا أن نسبق البقية في مضمار العلم والاكتشافات سيجبرون قصرا على تعلم اللغة العربية.

- وأيضا قبل كل ذلك أن يتم تعريب العلوم الموجودة، لربط أطفالنا وشبابنا (علماء المستقبل) باللغة العربية، والترسيخ في عقولهم المصلحات العربية بدلا من تلك الإنجليزية.

- وأيضا من أحدى الاقتراحات أن يتم تقديم منح لطلاب العلم من الناطقين باللغات الأجنبية ليدرسوا في بلادنا وتحفيزهم وحتى إن وصل الأمر أن يُدفع للنابغين منهم مبالغ مالية ليقدروا على تحمل نفقة العيش.

## ضوء خافت ينبثق

إنني حينما ابتدأت هذا المشروع، لم تكن المحاولة الأولى، إنما ذلك الشغل الشاغل لمحبي لغتهم والمعتزين بها، التي كانت لغة العلم وستعود بإذن الله. فقد وجدت بعضهم بدأ يوطئ للعلم حتى يسلك طريقه لمكانه الطبيعي.

- _**الرياضيات**_: التي كنت أدرسها ،في الأرض التي ولدت بها مصر، خلال فترة التعليم الأساسي، حيث كانت مميزة عن الفيزياء والكيمياء، محببة للقلب مع محبتنا للأخريين؛ فقلوبنا تهفوا للرموز العربية في المعادلات والحسابات الرياضيات.
- _**خط.سين**_: حيث قام "غسان السقاف" بعمله الرائع وتنفيذ الفكرة التي راودتني ففضفضت بها لصديق لي، ثم ببحث على جوجل ظهر لي موقعه ([خط](https://khatt.org/)). يمكنا القول أنه عرَّب بعضا من latex.
- _**لغة ألف**_: تشرفت بمعرفة أناس حرصهم على اللغة العربية دافع لهم في حياتهم، تبادلت معهم النقاشات والآراء حول اللغة الواعدة، [الصفحة الرئيسية](https://github.com/alifcommunity/).
- _**لغة خوارزم**_: [الصفحة الرئيسية](https://alkhawarizm.org/).
- و كثير من اللغات الأخرى التي كانت مهدها نفس مهد هذه المكتبة والتي لم يقدر الله لها أن تقدم أو تتقدم كثيرا، لأنها بدأت عملا فرديا وبقيت عملا فرديا. روي عن ابن عباس رضي الله عنهما عن النبي صلى الله عليه وسلم: "يد الله مع الجماعة".

## طريق وعر

> تنويه: هذه المشكلات تم رصدها في 1442

- من المشكلات ملف "اقرأني"، الذي يعرض في جيتهاب ومدير حزم نود npm، إذ يعرض بشكل أساسي مكتوبا بالاتجاه من اليسار لليمين، ومما هو مطلوب أن نقوم بتقديم طلب لهم أن يحلوا هذه المشكلة، وبطريقة غير مباشرة أن ننشر إضافة للمتصفحات (مثلا جوجل كروم) بحيث تقوم بتعديل اتجاه الكتاب إذا بدأ السطر بحرف من اللغة العربية. والحل الذي انتقيته الآن هو وضع كل سطور (markdown) داخل وسم ل.و.ت، (لغة الوسوم التشعبية HTML)، ووضع خاصيته `dir=rtl`.
- وأيضا نجد نفس المشكلة في الكود وخصوصا إن أردت الكتابة في "VSCode" وهو ما يمكن حله أيضا بعمل إضافات له لكي يدعم الكتابة من اليمين لليسار، إلى أن يقوموا هم بإتاحة الكتابة من اليمين لليسار.

- ومن العقبات أن الكون يصعب قراءته لأنه لا يحصل له تلوين "Syntax Hightlighting" كما في باقي اللغات والأكواد البرمجية ولذا يجب أن نقوم بعمل إضافة لمحررات الأكواد.

- ومشكلة أخرى تتعلق بطبيعة كتابة اللغة العربية، فما وجدته أن اللغات اللاتينية وما يكتب بحروفها تكتب بحروف مقطعة عند التعبير عن الكلمات، على عكس اللغة العربية. وما المشكلة؟ إنها تكمن في أن كافة فروع العلم وحتى المجالات غير العلمية والجوانب المختلفة من الحياة تعتمد على الكلمات المختصرة أو ما يدعى في الإنجليزية بـ "acronym". وحل هذا بسيط جدا وذلك بكتابة الحروف العربية منفصلة أيضا، أبهرتك أليس كذلك؟! لكن يمكننا أن نفعل هذا عند الكتابة بالقلم ولا يمكننا أن نقوم بهذا على الحاسوب أو الهاتف الذكي إلا بوضع مسافات أو أي حرف من غير محارف اللغة مثل النقطة ".". وأنا أقترح أن نجعل مفتاح"CAPS" هو ما يجعل الحروف مقطعة حين الكتابة والضغط عليه متزامنين، وهذا يتطلب إلغاء الوظيفة المعتادة له في اللغات اللاتينية وتوظيفه كما نريد في اللغة العربية.

## كيف نحل كل ذلك

لتحقيق الغاية لن نقف مكتوفي الأيدي تاركين الغرب مستحوذاً على تقديم الخدمات والمنتجات البرمجية أو غيرها، فإنه على المدى البعيد ستكون منتجاتنا فهي المتصدرة والتي تدعم اللغة العربية بشكل رئيسي ثم نعطي بعد ذلك الاهتمام لما بعدها من الروافد من اللغات، إن شاء الله.

**Comment by xximj** [comment]
السلام عليكم ورحمة الله وبركاته 
بارك الله فيك أخي محمد واسأل الله ان يجعل عملنا جميعاً خالصاً لوجهه الكريم.  
كان عندي اقتراح بسيط بخصوص هذا الجزء:
```
ومشكلة أخرى تتعلق بطبيعة كتابة اللغة العربية، فما وجدته أن اللغات اللاتينية وما يكتب بحروفها تكتب بحروف مقطعة عند التعبير عن الكلمات، على عكس اللغة العربية. وما المشكلة؟ إنها تكمن في أن كافة فروع العلم وحتى المجالات غير العلمية والجوانب المختلفة من الحياة تعتمد على الكلمات المختصرة أو ما يدعى في الإنجليزية بـ "acronym". وحل هذا بسيط جدا وذلك بكتابة الحروف العربية منفصلة أيضا، أبهرتك أليس كذلك؟! لكن يمكننا أن نفعل هذا عند الكتابة بالقلم ولا يمكننا أن نقوم بهذا على الحاسوب أو الهاتف الذكي إلا بوضع مسافات أو أي حرف من غير محارف اللغة مثل النقطة ".". وأنا أقترح أن نجعل مفتاح"CAPS" هو ما يجعل الحروف مقطعة حين الكتابة والضغط عليه متزامنين، وهذا يتطلب إلغاء الوظيفة المعتادة له في اللغات اللاتينية وتوظيفه كما نريد في اللغة العربية.
```

الاقتراح هو استخدام طريقة الخطاط نصري خطار في كتابة الحروف العربية. وهي طريقة بسيطة سماها العربية الموحدة أو Unified Arabic تعطي كل حرف من حروف الهجاء شكلاً واحداً فتكون كتابة الاختصارات واضحة ويصبح الكود أبسط في القراءة.  

[نصري خطار - ويكيبيديا](https://ar.m.wikipedia.org/wiki/نصري_خطار)  
[خط مرسوم بالعربية الموحدة](https://www.29lt.com/product/29lt-unified-arabic-neo-b/)
[نماذج للخطوط](https://www.29lt.com/wp-content/uploads/2021/04/29LT_UA_Neo.pdf)

![image](https://user-images.githubusercontent.com/72635080/178162868-ca1283ed-37d5-4cd1-a50d-9368bcf421ae.jpeg)  

![image](https://user-images.githubusercontent.com/72635080/178162915-96af551c-1ae4-4dad-aa46-0659fd4b4d55.png)  

![image](https://user-images.githubusercontent.com/72635080/178162923-1c7d925a-8954-473a-bf26-59a74b6b8a2e.png)  

![image](https://user-images.githubusercontent.com/72635080/178162935-a277690f-2785-4dcc-8af4-3f3b0263e1ce.gif)

**Reply by Assayyaad** [reply]
تركيب الحروف مما يميز العربية عن غيرها من اللغات وإزالة هذا التمييز للخوض في ساحة اللغات الأخرى أراه خطأ، وتقطيع الكلمات واختصارها أصلا ليس من أصل العربية وإنما هو من طبيعة اللغة الإنجليزية وقد بُنيت الأجهزة والشبكات بناء على هذا، فبدلا من تشويه العربية لتتماشى مع البيئة التي بنيت للغة الإنجليزي علينا بناء بيئة بديلة مخصصة للغة العربية تحترم خصائصها وتقويها

**Reply by assawalhy** [reply]
نريد استخدام الاختصارات وسط الكلام العادي فلو قلت مثلا، سأتعلت ال ذص سيفهم الناس مثلا أني أريد تعلم الذكاء الاصطناعي
I want to learn AI
I want to learn artificial intelligence

هل رسم الاختصارات بطريقة رسم الكلمة العربية سيكون مناسبا؟ فاختصار مثل فقك (ألفته الآن عشوائيا)، هل سيفهمه الناس أنه اختصار أم كلمة؟ هنا سيصعب فهم الكلام المراد والحل المتاح الآن أن نفرق بين الحروف بعلامات مثلا (ف ق ك، ف.ق.ك، ف-ق-ك، ...)

**Reply by Assayyaad** [reply]
أنا ضد الإختصارات في العربية جملة وتفصيلا بغض النظر عن الرسم، والعلوم قديما مثل الرياضيات لما كتبها الخوارزمي كلها كانت بكلمات عربية بسيطة يفهمها الجميع

الرموز والإختصار دخيل أطل برأسه علينا مع الإستعمار وغرس جذوره في تعاملاتنا اليومية مع ثقافة المستعمر

والله أعلم

**Reply by Mohamed7AbdELhalim** [reply]
السلام عليكم ورحمة الله وبركاته 
### بما ان الشبكات وأسس الحاسوب بنيت بهذا الشكل فلدي سؤال له جانبين:
- الأول:
 هل نقوم بأستخدام الرمز أو المكتبة التي تدعم اللغة العربية ونقوم بترجمتها أم بناء نسخة جديدة بسياق ونسق ناوية بما يتناسب مع أسلوب اللغة العربية مثل الشعر والنثر والابيات، لها طريقة في الكتابة.

- الثاني:
بناء الحاسوب من الصفر مثلما قال الشارخ لتحقيق الهدف المنشود ويجتمع فيها من يتقن العربية ومن لا يتقن ونعلمهم أياها ونقوم بالبحث العلمي والتطوير سواء في مجموعة أو مؤسسات تقوم بهذا وليس مجموعة واحدة فقط. 

![Screenshot_٢٠٢٥٠٨١٨-٠١٣٤٠٧_Brave.jpg](https://github.com/user-attachments/assets/9770b6e0-539c-4c03-b8e7-4ee8317afbee)
صورة توضيحية لأحد النشرات الإخبارية.

فما رأيك هل نبني أسس جديدة أم ماذا؟


> تركيب الحروف مما يميز العربية عن غيرها من اللغات وإزالة هذا التمييز للخوض في ساحة اللغات الأخرى أراه خطأ، وتقطيع الكلمات واختصارها أصلا ليس من أصل العربية وإنما هو من طبيعة اللغة الإنجليزية وقد بُنيت الأجهزة والشبكات بناء على هذا، فبدلا من تشويه العربية لتتماشى مع البيئة التي بنيت للغة الإنجليزي علينا بناء بيئة بديلة مخصصة للغة العربية تحترم خصائصها وتقويها

**Reply by Mohamed7AbdELhalim** [reply]
بسم الله الرحمن الرحيم 
النقطة التي أشرتها سهلة لكن المشكلة أن اللغة العربية سوف تفقد معناها فاللغة نفسها لم تكتب في أي كتاب تاريخي بشكل منفصل إلا في الرياضيات فكون تنسيق ذلك على أنها أحرف غير مركبة سوف تفقد المعنى اللغوي في العربية.


> نريد استخدام الاختصارات وسط الكلام العادي فلو قلت مثلا، سأتعلت ال ذص سيفهم الناس مثلا أني أريد تعلم الذكاء الاصطناعي
> I want to learn AI
> I want to learn artificial intelligence

> هل رسم الاختصارات بطريقة رسم الكلمة العربية سيكون مناسبا؟ فاختصار مثل فقك (ألفته الآن عشوائيا)، هل سيفهمه الناس أنه اختصار أم كلمة؟ هنا سيصعب فهم الكلام المراد والحل المتاح الآن أن نفرق بين الحروف بعلامات مثلا (ف ق ك، ف.ق.ك، ف-ق-ك، ...)

**Reply by assawalhy** [reply]
> أنا ضد الإختصارات في العربية جملة وتفصيلا بغض النظر عن الرسم، والعلوم قديما مثل الرياضيات لما كتبها الخوارزمي كلها كانت بكلمات عربية بسيطة يفهمها الجميع
> 
> الرموز والإختصار دخيل أطل برأسه علينا مع الإستعمار وغرس جذوره في تعاملاتنا اليومية مع ثقافة المستعمر
> 
> والله أعلم

أتفق لكن لا مشكلة من إيجاد حلول لمشاكل جديدة تواجهها اللغة لما توصل إليه البشر من تقدم علمي أو تغير في طريقة العيش
فمثلا اللغة العربية لم يكن بها نقاط وعلم النحو وغيره من علوم اللغة استحدثوا بعد عصور البلاغة العربية لما واجهه الناس من تحديات بسبب دخول العجم في الإسلام وكلامهم بالعربية ولحنهم في اللغة فكان لا بد من وضع أصول وقواعد للغة لنعلمها لهم ولضبط لسان العرب أنفسهم.

يمكن أن لا يكون تقطيع الكلمات واختصارها في حروف حلا سليما يناسب العربية لكن ربما نجد حلا غيره إن شاء الله.

**Comment by Mohamed7AbdELhalim** [comment]
وعليكم السلام ورحمة الله
أجد صعوبة في القراءة وأيضا سوف يكون أصعب أن طبقت بشكل فعلي في الكود فأن كان ما فهمت صحيح فأن المبرمج سوف يتعب من قراءة الكود عندما يراجعها.


> السلام عليكم ورحمة الله وبركاته 
> بارك الله فيك أخي محمد واسأل الله ان يجعل عملنا جميعاً خالصاً لوجهه الكريم.  
> كان عندي اقتراح بسيط بخصوص هذا الجزء:
> ```
> ومشكلة أخرى تتعلق بطبيعة كتابة اللغة العربية، فما وجدته أن اللغات اللاتينية وما يكتب بحروفها تكتب بحروف مقطعة عند التعبير عن الكلمات، على عكس اللغة العربية. وما المشكلة؟ إنها تكمن في أن كافة فروع العلم وحتى المجالات غير العلمية والجوانب المختلفة من الحياة تعتمد على الكلمات المختصرة أو ما يدعى في الإنجليزية بـ "acronym". وحل هذا بسيط جدا وذلك بكتابة الحروف العربية منفصلة أيضا، أبهرتك أليس كذلك؟! لكن يمكننا أن نفعل هذا عند الكتابة بالقلم ولا يمكننا أن نقوم بهذا على الحاسوب أو الهاتف الذكي إلا بوضع مسافات أو أي حرف من غير محارف اللغة مثل النقطة ".". وأنا أقترح أن نجعل مفتاح"CAPS" هو ما يجعل الحروف مقطعة حين الكتابة والضغط عليه متزامنين، وهذا يتطلب إلغاء الوظيفة المعتادة له في اللغات اللاتينية وتوظيفه كما نريد في اللغة العربية.
> ```
> 
> الاقتراح هو استخدام طريقة الخطاط نصري خطار في كتابة الحروف العربية. وهي طريقة بسيطة سماها العربية الموحدة أو Unified Arabic تعطي كل حرف من حروف الهجاء شكلاً واحداً فتكون كتابة الاختصارات واضحة ويصبح الكود أبسط في القراءة.  
> 
> [نصري خطار - ويكيبيديا](https://ar.m.wikipedia.org/wiki/نصري_خطار)  
> [خط مرسوم بالعربية الموحدة](https://www.29lt.com/product/29lt-unified-arabic-neo-b/)
> [نماذج للخطوط](https://www.29lt.com/wp-content/uploads/2021/04/29LT_UA_Neo.pdf)
> 
> ![image](https://user-images.githubusercontent.com/72635080/178162868-ca1283ed-37d5-4cd1-a50d-9368bcf421ae.jpeg)  
> 
> ![image](https://user-images.githubusercontent.com/72635080/178162915-96af551c-1ae4-4dad-aa46-0659fd4b4d55.png)  
> 
> ![image](https://user-images.githubusercontent.com/72635080/178162923-1c7d925a-8954-473a-bf26-59a74b6b8a2e.png)  
> 
> ![image](https://user-images.githubusercontent.com/72635080/178162935-a277690f-2785-4dcc-8af4-3f3b0263e1ce.gif)
>
